protocol/0078-NWLI-network_wide_limits.md

```diff
@@ -0,0 +1,128 @@
+
+## Parties in a market
+
+Introduce a new network parameter `limits.markets.maxParties` controlling the maximum number of parties allowed within any given market. Default value: `100,000`.
+
+A party gets counted towards the limit if it has either open orders or open positions in the market. Once party has no open orders and no open positions it gets removed from a **total party count** within the market. Once the limit gets reached the market accepts no further orders from parties that are not already in the market.
+Once the total party count drops below the limit market accepts orders from any party again (provided its [state](./0043-MKTL-market_lifecycle.md#market-lifecycle-statuses) allows that).
+
+The limit does not apply to liquidity providers.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.\
+If it gets decreased below the current total party count in any given market then no further actions are needed. Specifically: all parties in that market remain unaffected (their positions and orders remain open), but no new parties are allowed to participate in the market until the total party count drops below the new limit.
+
+## Limit orders on a book
+
+Introduce a new network parameter `limits.markets.maxLimitOrders` controlling the maximum number of limit orders that can rest on a book in any given market.\
+Default value: `1,000,000`.
+
+A limit order of arbitrary volume which gets placed on the book (doesn't trade in full on entry) contributes +1 to the count. If an order already on the book gets cancelled or filled in full (so that it's remaining size is 0) the count should be decreased by 1. If the count reaches `limits.markets.maxLimitOrders` limit orders can still be submitted, but:
+
+* if the order is aggressive (results in a trade) the trade proceeds as normal,
+* if the order or its part is passive (it doesn't match in full on entry to the matching engine) and it's part would get added to the order book (it's a persistent order) that part of the order gets rejected.
+
+Pegged orders and [liquidity provision orders](./0038-OLIQ-liquidity_provision_order_type.md) do not get counted towards the limit.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.\
+If it gets decreased below the current total number of orders on the book in any given market then no further actions are needed. Specifically: all open orders in that market remain unaffected, but no new orders are allowed until the **total open order count** drops below the new limit.
+
+## Pegged orders on a market
+
+Introduce a new network parameter `limits.markets.maxPeggedOrders` controlling the maximum number of pegged orders that can be active in any given market.\
```

Comment:
Shouldn't that be a limit per party? it is so low that it seems that a few parties could easily reach that point?